Examples/Algorithms/Geant4/src/SensitiveSteppingAction.cpp

```diff
@@ -185,9 +202,57 @@ void SensitiveSteppingAction::UserSteppingAction(const G4Step* step) {
     return;
   }
 
+  // Output is only strictly valid if step logging is not enabled
   const auto particleId = eventStore().trackIdMapping.at(track->GetTrackID());
-
-  ACTS_VERBOSE("Step of " << particleId << " in sensitive volume " << geoId);
+  if (!m_cfg.stepLogging && surface != nullptr) {
+    ACTS_VERBOSE("Step of " << particleId << " in sensitive volume " << geoId);
+  } else if (m_cfg.stepLogging) {
+    if (!eventStore().propagationRecords.contains(track->GetTrackID())) {
+      // Create the propagation summary
+      double xVtx = track->GetVertexPosition().x();
+      double yVtx = track->GetVertexPosition().y();
+      double zVtx = track->GetVertexPosition().z();
+      double xDirVtx = track->GetVertexMomentumDirection().x();
+      double yDirVtx = track->GetVertexMomentumDirection().y();
+      double zDirVtx = track->GetVertexMomentumDirection().z();
+      double absMomentum = track->GetMomentum().mag();
+
+      PropagationSummary iSummary(Acts::CurvilinearTrackParameters(
+          Acts::Vector4(xVtx, yVtx, zVtx, 0.),
+          Acts::Vector3(xDirVtx, yDirVtx, zDirVtx), absCharge / absMomentum,
+          std::nullopt, Acts::ParticleHypothesis::pion()));
```

Comment:
do we have to worry about units here?

Reply:
Good question, will look at it.

Reply:
indeed this was missing.

Reply:
don't we have to use conversions similar to the ones above to get the right quantities for the propagation summary?

```c++
static constexpr double convertLength = Acts::UnitConstants::mm / CLHEP::mm;
static constexpr double convertEnergy = Acts::UnitConstants::GeV / CLHEP::GeV;
static constexpr double convertTime = Acts::UnitConstants::ns / CLHEP::ns;
```

Reply:
I missed the energy .. the other one is a direction  ... will fix.

Reply:
Updated.